Examples/HelloWorld/CMakeLists.txt

```diff
@@ -6,3 +6,45 @@ target_link_libraries ( HelloWorld ${SimpleITK_LIBRARIES} )
 if ( WRAP_CSHARP )
   csharp_add_executable( HelloWorldCS SimpleITKCSharpManaged.dll ${CMAKE_CURRENT_SOURCE_DIR}/HelloWorld.cs )
```

Comment:
Using `sitk_add_csharp` would make it consistent with the other two csharp examples and java. the C++ does appears to always get compiled though.

I am not sure the behavior of `BUILD_TESTING` is consistent. The behavior could be make consistent inside the `sitk_add_java_test` and `sitk_add_csharp_test`. Please remove the separate sharp compilation, and use the macro. The logic for building the tests can be modified in another patch inside these  testing macros.